src/vcpkg/commands.build.cpp

```diff
@@ -1756,121 +1756,55 @@ namespace vcpkg
                                const std::unordered_map<std::string, std::string>& cmakevars)
         : triplet(triplet), m_paths(paths)
     {
-        enum class VcpkgTripletVar
-        {
-            TARGET_ARCHITECTURE = 0,
-            CMAKE_SYSTEM_NAME,
-            CMAKE_SYSTEM_VERSION,
-            PLATFORM_TOOLSET,
-            PLATFORM_TOOLSET_VERSION,
-            VISUAL_STUDIO_PATH,
-            CHAINLOAD_TOOLCHAIN_FILE,
-            BUILD_TYPE,
-            ENV_PASSTHROUGH,
-            ENV_PASSTHROUGH_UNTRACKED,
-            PUBLIC_ABI_OVERRIDE,
-            LOAD_VCVARS_ENV,
-            DISABLE_COMPILER_TRACKING,
-            XBOX_CONSOLE_TARGET,
-            Z_VCPKG_GameDKLatest
-        };
+        Util::assign_if_set_and_nonempty(target_architecture, cmakevars, CMakeVariableTargetArchitecture);
+        Util::assign_if_set_and_nonempty(cmake_system_name, cmakevars, CMakeVariableCMakeSystemName);
+        Util::assign_if_set_and_nonempty(cmake_system_version, cmakevars, CMakeVariableCMakeSystemVersion);
+        Util::assign_if_set_and_nonempty(platform_toolset, cmakevars, CMakeVariablePlatformToolset);
+        Util::assign_if_set_and_nonempty(platform_toolset_version, cmakevars, CMakeVariablePlatformToolsetVersion);
+        Util::assign_if_set_and_nonempty(visual_studio_path, cmakevars, CMakeVariableVisualStudioPath);
+        Util::assign_if_set_and_nonempty(external_toolchain_file, cmakevars, CMakeVariableChainloadToolchainFile);
+        if (auto value = Util::value_if_set_and_nonempty(cmakevars, CMakeVariableBuildType))
+        {
+            if (Strings::case_insensitive_ascii_equals(*value, "debug"))
+                build_type = ConfigurationType::Debug;
+            else if (Strings::case_insensitive_ascii_equals(*value, "release"))
+                build_type = ConfigurationType::Release;
+            else
+                Checks::msg_exit_with_message(VCPKG_LINE_INFO, msgUnknownSettingForBuildType, msg::option = *value);
+        }
```

Comment:
this looks like `value_if_set_and_valid`

Reply:
This would be the same as optional. See my comment above.